json_diff_test.go

```diff
@@ -246,9 +244,12 @@ func TestShouldReturnFielsdWhenMoreThanOneFieldAreDifferent(t *testing.T) {
 		"num":   2,
 	}
 
+	expected := []string{"array", "json", "num", "str"}
 	actual := Diff(value, value2)
```

Comment:
No love for using a test package separated from the actual package ? I enjoy it because it prevents coupling between the tests and the code under test and also provides a better feel on how it will be to use the code since you import and use it just as the client will do.

Reply:
I agree! Will create another PR for that :)